docs/libvirt-pod-networking.md

```diff
@@ -0,0 +1,118 @@
+a minimal integrated solution, with few workarounds.
+
+**Note:** This approach is rough, and not optimized, but the rule of thumb was
+to find a robust solution which works with Kubernetes today.
+
+
+## Use-case
+
+The primary use-case is to allow VM to get connected to the pod network.
+
+
+## API
+
+The API is pretty clear: As usual an interface will be defined.
+To indicate that Pod interfaces shall be used for networking, a new type
+`pod-network` is introduced.
+All other fields, concerning the properties of the virtual NIC, can be used
+as usual.
+
+```yaml
+kind: VM
+spec:
+  domain:
+    devices:
+      interfaces:
+      - type: pod-network
+        mac:
+          address: 00:11:22:33:44:55
+        model:
+          type: virtio
+```
+
+
+## Implementation
+
+Prerequisites
+- libvirt is using Pod networking
+- A CNI proxy to call the hosts CNI is available (this can be part of this
+  designs implementation)
+
+Conceptually the implementation works by assuming that every vNIC will be
+mapped to a _new_ Pod interface. To achieve this, for every vNIC a _new_ Pod
+interface is requested through CNI.
+Any after creation of the Pod created interfaces will from now on be called
+_VM interface_ to differentiate them from the the original during pod creation
+allocated Pod interface.
+
+**Note:** The original Pod interface (eth0) is never touched and can be used to
+access libvirtd (through the libvirtd Pod IP) or to provide VM centric services
+through the VM Pod IP.
+
+**Note:** The _new_ VM interface can not be directly attached to the VM,
+because we lose the ability to provide the IP to the guest. The IP can only be
+provided by DHCP, which requires the use of a bridge.
+
+The broader steps to enable networking in the described way:
+
+1. Request/Create a _new_ VM interface on the libvirt pod, by using the CNI
+   proxy to use the hosts CNI configuration and plugins.
+2. Remember the IP of the VM interface, then remove it from the VM interface
+3. Create a libvirt network, backed by a bridge, enable DHCP,
+   and add a single host entry, with the remembered IP address, and the MAC
+   address of the corresponding virtual NIC of the VM
+4. On shutdown, the to be deleted VM interfaces can be infered by
+   following the networks associated with the vNICs
+
+This process must be repeated for every virtual NIC of every VM.
+Thus N virtual NICs correspond to N VM interfaces, and in turn to N IP
+addresses.
+
+**Note:** This might look like an overhead, creating a single VM interface
+per virtual NIC. The reason for doing so is, that this is a Kubernetes/CNI
+compatible way to request IP endpoints from the networking sub-system.
+
+
+## Drawbacks & Limitations
+
+* Every virtual NIC can only use a single IP address (the one provided via
+  DHCP)
+* A new VM interface is required for every vNIC.
+* Cleanup of the VM interfaces in case of an uncontrolled pod shutdown is
```

Comment:
but we are not sure that your current text is correct. How about a softer statement:

"the added VM interfaces are properly cleaned by docker in case on an uncontrolled shutdown of the libvirt pod"